crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2380,11 +2416,84 @@ impl<'db> TypeInferenceBuilder<'db> {
             ctx: _,
         } = subscript;
 
-        self.infer_expression(slice);
-        self.infer_expression(value);
-
-        // TODO actual subscript support
-        Type::Unknown
+        let value_ty = self.infer_expression(value);
+        let slice_ty = self.infer_expression(slice);
+
+        match (value_ty, slice_ty) {
+            // TODO handle variable-length tuples
+            (Type::Tuple(tuple_ty), Type::IntLiteral(int)) => {
+                let elements = tuple_ty.elements(self.db);
+                usize::try_from(int)
```

Comment:
Python allows negative indexing, which indexes from the end of the container:

```
>>> t = (1, 2, 3)
>>> t[-1]
3
```

So we should handle that here. I would say we don't have to do it in this PR, and we can just add a TODO, but I don't like that in the meantime this would mean we'd wrongly emit an out-of-bounds error.

Reply:
I'll handle it here, thanks.